### DISCUSSION TITLE
Support for Nuxt 3 (to really compete with Strapi)

### DISCUSSION THREAD

**Discussion by casualmatt** [directus/directus#11649] [Feature Requests]
Hello, 

I'm a Strapi and Directus user, and i love how both are different in many aspect, so that i can pick the best option for the project, especially because i'm mostly a nocoder for the backend. 

But for the frontend i would like to be able to code and customize my UI/UX with some code. 
This currently is not really possible with Directus because the SDK is not integrated with any Front-end Framework to complete what could be a perfect Jamstack. 

Most of the time i have to fight to integrate Directus with Nuxt, because the only [Plugin for Nuxt (2) is not official](https://github.com/pnodev/nuxt-directus) (and now unmaintained), while if I pick Strapi i have support for Nuxt 2, bridge and 3, but then i don't have all the feature that Directus offer. 

I would like to see a porting or the SDK as a Nuxt Brige or just only Nuxt 3 plugin. 

My example here is with Nuxt because is the biggest Vue full framework available, and Vue is the front end framework used by Directus.

**Comment by benhaynes** [comment]
Thoughts, @rijkvanzanten?

**Reply by casualmatt** [reply]
[Prismic](https://v3.prismic.nuxtjs.org/) also publish his plugin, and [Storyblok ](https://github.com/storyblok/storyblok-nuxt)was publish to.

**Reply by benhaynes** [reply]
I'm not technical enough to keep up here, but are you essentially looking for an integration guide?
CC @azrikahar

**Reply by casualmatt** [reply]
Can be both, a integration from the SDK with Nuxt (AKA a plugin) or guide on how to use and how you think should be use your SDK with the Nuxt3 features.

**Reply by licitdev** [reply]
I believe this discussion is related to PR https://github.com/directus/directus/pull/11668 by @azrikahar

**Reply by casualmatt** [reply]
Yes it is exactly that, ty @licitdev 

I think/hope that @azrikahar is developing for nuxt 3, even if it's still in beta.

**Reply by azrikahar** [reply]
> Most of the time i have to fight to integrate Directus with Nuxt

As @rijkvanzanten pointed out, technically you can already use Directus SDK in a plugin, [provide it as a helper](https://v3.nuxtjs.org/docs/directory-structure/plugins#automatically-providing-helpers) (Nuxt 3) or [inject it to the context](https://nuxtjs.org/docs/directory-structure/plugins/#inject-in-root--context) (Nuxt 2) and it will work out of the box. 

> I think/hope that @azrikahar is developing for nuxt 3, even if it's still in beta.

The PR itself does have a checklist 👍 The tentative idea for now is developing it for Nuxt 3 with the recent new module builder, then likely a guide on the plugin approach mentioned above for any incompatible versions.

**Reply by casualmatt** [reply]
> [provide it as a helper](https://v3.nuxtjs.org/docs/directory-structure/plugins#automatically-providing-helpers) (Nuxt 3)

that's true, but it's in general a easier dev experience having an official module, with some example and some documentation, exactly like this[ un-official plugin for v2. 
](https://github.com/pnodev/nuxt-directus)

**Reply by rijkvanzanten** [reply]
> > [provide it as a helper](https://v3.nuxtjs.org/docs/directory-structure/plugins#automatically-providing-helpers) (Nuxt 3)
> 
> that's true, but it's in general a easier dev experience having an official module, with some example and some documentation, exactly like this[ un-official plugin for v2.](https://github.com/pnodev/nuxt-directus)

Good thing we just launched [the guides section on the website](https://directus.io/guides/) and [the examples repo](https://github.com/directus/examples) 😄 Perfect place to add in an article/example for Nuxt 3 👍🏻

**Reply by casualmatt** [reply]
> [provide it as a helper](https://v3.nuxtjs.org/docs/directory-structure/plugins#automatically-providing-helpers) (Nuxt 3)

Hello @azrikahar i try it, to import it as a Helper but it doesn't work. 
Maybe can you update your examples repo, with one Nuxt 3 install.

**Comment by rijkvanzanten** [comment]
Just for the record, you don't _need_ the Nuxt 3 plugin in order to use Directus with Nuxt 🙂 The regular JavaScript SDK works great with Nuxt 3 as well!

**Reply by Intevel** [reply]
But the DX with a module is better.

**Comment by casualmatt** [comment]
Got publish yesterday. 
 
https://github.com/intevel/nuxt-directus // https://nuxt-directus.netlify.app/usage/useDirectusItems

**Reply by Intevel** [reply]
Yes, I started developing the module before I saw that Directus was working on a module. I am trying to get in touch with the Directus team to talk about it.

**Reply by Nisthar** [reply]
@casualmatt i am curious as to what are the features that directus have that are not in strapi?

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
